tests/test_transaction.py

```diff
@@ -117,6 +117,37 @@ def test_transaction_commit(db, col, docs):
     assert err.value.error_code in {10, 1655}
 
 
+def test_transaction_fetch_existing(db, col, docs):
+    original_txn = db.begin_transaction(
+        read=col.name,
+        write=col.name,
+        exclusive=[],
+        sync=True,
+        allow_implicit=False,
+        lock_timeout=1000,
+        max_size=10000,
+    )
+    txn_col = original_txn.collection(col.name)
+
+    assert "_rev" in txn_col.insert(docs[0])
+    assert "_rev" in txn_col.delete(docs[0])
+
+    txn_db = db.fetch_transaction(transaction_id=original_txn.transaction_id)
+
```

Comment:
Nice test! I suggest adding another explicit check to make sure the transaction_id property stays the same.

```python
assert original_txn.id == txn_db.id
```

Reply:
Agreed! Done.